### DISCUSSION TITLE
Have I bricked my ESP32?

### DISCUSSION THREAD

**Discussion by MartinDavidWaller** [nanoframework/Home#1522] [General]
Hi, I've just started playing with the nanoFramework and all was going well. Then the blue LED on my EXP32 DevKit came on and deployments now fail. I can execute the command:

nanoff --platform esp32 --serialport COM5 --update

from a cmd window and all is well. After that the blue LED comes an and Device Explorer does not see the ESP32. I've tried rebooting, restarting VS, etc but nothing seems to work!

Any advice greatly appreciated.

Martin

**Comment by MartinDavidWaller** [comment]
The command:

nanoff --listdevices  -v diag

simply returns:

No devices found.

**Comment by Ellerbach** [comment]
it's almost impossible to brick an ESP32 :-) And definitely not with nanoFramework.
Try to add `--masserase` in the command line to flash. And reflash the device.
Also unplug and replug the device just to check.
And some devices have a reset button you may have to press and release, a reboot button, they sometimes need to be combined.
And you have a full guide here to help you: https://docs.nanoframework.net/content/getting-started-guides/troubleshooting-device-connection.html

**Reply by MartinDavidWaller** [reply]
Hi Laurent,

Many thanks for getting back to me with this. I've tried your suggestion and, as yet, no luck. I tried for an hour executing the:

nanoff --platform esp32 --serialport COM5 --update

command both with and without the --masserase option. 

I pointed putty at the device and watched it boot. It doesn't display any errors but it looks like it's still executing the last code that I upload. That code does a lot of output using Debug.WriteLine. 

Out of interest I used to Arduino UDE to upload a scetch to the device and that all worked. Using Putty I could see the output I expected. I then re executed the:

nanoff --platform esp32 --serialport COM5 --update

command and no luck with Visual Studio device manager. I then ran Putty again and it look like it is still running the old code!

Interesting!

Martin

**Reply by Ellerbach** [reply]
make sure you definitely add the `--masserase` flag. If you've uploaded something else but did not dleployment partition, indeed, what's left is left. So if you deploy anything else, from any other plaform, when back to nano, use that flag.

**Comment by MartinDavidWaller** [comment]
Hi Laurent,

I have it going now. From another thread someone suggested:

C:\Users\marti\.dotnet\tools\.store\nanoff\2.5.90\nanoff\2.5.90\tools\net8.0\any\esptool\esptoolWin>esptool --chip esp32 erase_flash

And that seemed to make the difference.

Many thanks for your help

Martin
